conans/test/functional/toolchains/cmake/test_cmake_toolchain_m1.py

```diff
@@ -64,8 +64,6 @@ def build(self):
     assert "CMAKE_SYSTEM_PROCESSOR: arm64" in client.out
     main_path = "./build/Release/main.app/main" if op_system == "iOS" \
         else "./build/Release/main"
-    client.run_command(main_path, assert_error=True)
```

Comment:
Can these asserts be kept checking the architecture?

Reply:
(adding this to keep it on the records, since we discussed it in person)

The architecture of the built file is already checked below:
```
    client.run_command("lipo -info {}".format(main_path))
    assert "Non-fat file" in client.out
    assert "is architecture: arm64" in client.out
```

So running it to confirm that the "bad CPU type in executable" is printed, as well as checking the architecture by querying the output of the `lipo` tool would appear as two different ways of checking the same post-condition. 

However we could indeed keep the assert conditional.

On a separate now, since this test is checking that we can successfully build a binary for iOS with sdk 12.0 as the base target, I've added a couple of asserts checking that (since it's still possible to accidentally build it for macos)